### DISCUSSION TITLE
Support for HasOneThrough in FilamentPHP 3.0

### DISCUSSION THREAD

**Discussion by kaitoAckerman** [filamentphp/filament#9000] [Help]
Hi there, 

Recently i've migrated from Filament v2 into Filament v3. But noticed, hasOneThrough relationship seems not working on filter code.

```php
SelectFilter::make('vendor')->relationship('vendor', 'name')
```

This is relationship code
```php
public function vendor()
    {
        return $this->hasOneThrough(
            Vendor::class,
            Package::class,
            'id',
            'id',
            'package_id',
            'vendor_id'
        );
    }
```

This code will return an error

```
Filament\Forms\Components\Select::getRelationship(): Return value must be of type Illuminate\Database\Eloquent\Relations\BelongsTo|Illuminate\Database\Eloquent\Relations\BelongsToMany|Znck\Eloquent\Relations\BelongsToThrough|null, Illuminate\Database\Eloquent\Relations\HasOneThrough returned
```

help please ?

**Comment by samuel-fonseca** [comment]
Even though you posted this a while ago, but I actually came across this earlier today. The solution I found is to key the all relationships, in the example you provided, you can use:

```php
SelectFilter::make('vendor')->relationship('package.vendor', 'name');
```

I am curious as to why the underlying logic for the `FormBuilder` does not support this by default.
